src/main/resources/liquibase/202305031403-create-subscription-measurements-table.xml

```diff
@@ -0,0 +1,229 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<databaseChangeLog
+  xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+
+  <property name="update_statement_for_has_unlimited_usage" dbms="postgresql"
+    value="UPDATE subscription s SET has_unlimited_usage = sc.has_unlimited_usage FROM
+      subscription_capacity sc WHERE sc.subscription_id = s.subscription_id" />
+
+  <property name="update_statement_for_has_unlimited_usage" dbms="hsqldb"
+    value="MERGE INTO subscription USING subscription_capacity
+    ON subscription.subscription_id = subscription_capacity.subscription_id
+    WHEN MATCHED THEN UPDATE
+    SET subscription.has_unlimited_usage = subscription_capacity.has_unlimited_usage"/>
+
+  <changeSet id="202305031403-01" author="awood">
+    <createTable tableName="subscription_measurements">
+      <column name="subscription_id" type="VARCHAR(255)"/>
+      <column name="start_date" type="TIMESTAMP WITH TIME ZONE"/>
+      <column name="metric_id" type="VARCHAR(255)"/>
+      <column name="measurement_type" type="VARCHAR(255)"/>
+      <column name="value" type="DOUBLE PRECISION"/>
+    </createTable>
+    <addPrimaryKey constraintName="subscription_measurements_pk"
+      tableName="subscription_measurements"
+      columnNames="subscription_id, start_date, metric_id,measurement_type"/>
+    <addForeignKeyConstraint constraintName="subscription_fk"
+      baseTableName="subscription_measurements"
+      referencedTableName="subscription"
+      baseColumnNames="subscription_id, start_date"
+      referencedColumnNames="subscription_id, start_date"/>
+  </changeSet>
+
+  <changeSet id="202305031403-02" author="awood">
+    <addColumn tableName="subscription">
+      <column name="has_unlimited_usage" type="BOOLEAN"/>
+    </addColumn>
+  </changeSet>
+
+  <changeSet id="202305031403-03" author="awood">
+    <createTable tableName="subscription_product_ids">
+      <column name="product_id" type="VARCHAR(32)"/>
+      <column name="subscription_id" type="VARCHAR(255)"/>
+      <column name="start_date" type="TIMESTAMP WITH TIME ZONE"/>
+    </createTable>
+    <addPrimaryKey constraintName="subscription_product_ids_pk"
+      tableName="subscription_product_ids"
+      columnNames="product_id, subscription_id, start_date"/>
+    <addForeignKeyConstraint constraintName="subs_product_ids_fk"
+      baseTableName="subscription_product_ids"
+      referencedTableName="subscription"
+      baseColumnNames="subscription_id, start_date"
+      referencedColumnNames="subscription_id, start_date"/>
+  </changeSet>
+
+  <!-- NB: All the changesets dealing with functions and triggers are scoped to postgresql.  We
+  don't need them during unit tests -->
```

Comment:
I dislike the added complexity of the triggers/functions. Given we do a nightly sync of subscriptions, I'd rather remove them and let any inconsistencies be addressed by the next subscription sync.

Reply:
+1

Reply:
After some discussion, we're leaning towards keeping them (but they won't be long lived)